bl1/aarch64/bl1_exceptions.S

```diff
@@ -26,25 +26,25 @@ vector_entry SynchronousExceptionSP0
 	mov	x0, #SYNC_EXCEPTION_SP_EL0
 	bl	plat_report_exception
 	no_ret	plat_panic_handler
-	check_vector_size SynchronousExceptionSP0
+end_vector_entry SynchronousExceptionSP0
```

Comment:
minor: It would be good if the commit message explained why this new macro is needed (i.e. that the Clang linker cannot resolve the calculation in check_vector_size), Also that this macro now doesn't do anything and is deprecated.